### PR TITLE
Add parameter LimitResultPerPart to filter classes

### PR DESCRIPTION
**Pull request by czthiele**
New parameter for measurement search: LimitResultPerPart.
- in combination with a part path and deep this will trigger a search for all child parts and return the specified amount of measurements per part
- in combination with a list of part uuids this will return the specified amount of measurements per part uuid

### CODE REVIEW

**Review by czkum — CHANGES_REQUESTED** [review]

File: src/Api.Rest/Contracts/DataServiceFeatureMatrix.cs [hunk 1]
```diff
@@ -52,6 +52,9 @@ public class DataServiceFeatureMatrix : FeatureMatrix
 		//Request asynchronous deletion of measurements and check status with long polling.
 		public static readonly Version AsyncMeasurementDeletionMinVersion = new Version( SupportedMajorVersion, 7 );
 
+		//Request a limit per part when fetching measurements.
```
[the remark this answers is not in the document]
Comment (czthiele) [reply src/Api.Rest/Contracts/DataServiceFeatureMatrix.cs]: Done

File: src/Api.Rest/HttpClient/Data/DataServiceRestClient.cs [hunk 2]
```diff
@@ -304,6 +304,21 @@ private async Task DeleteMeasurementsInternal( List<ParameterDefinition> paramet
 			}
 		}
 
+		private async Task CheckLimitResultPerPartFeatureSupport( AbstractMeasurementFilterAttributesDto filter, CancellationToken cancellationToken )
```
[the remark this answers is not in the document]
Comment (czthiele) [reply src/Api.Rest/HttpClient/Data/DataServiceRestClient.cs]: Good point, I changed the FeatureMatrix check methods to this name pattern.

File: src/Api.Rest/HttpClient/Data/DataServiceRestClient.cs [hunk 3]
```diff
@@ -727,6 +742,8 @@ public async Task<SimpleMeasurementDto[]> GetMeasurements( PathInformationDto pa
 				}
 			}
 
+			await CheckLimitResultPerPartFeatureSupport( filter, cancellationToken );
```
[the remark this answers is not in the document]
Comment (czthiele) [reply src/Api.Rest/HttpClient/Data/DataServiceRestClient.cs]: I have moved all "feature checks" to separate methods which makes the code more readable.

File: src/Api.Rest/HttpClient/Data/DataServiceRestClient.cs [hunk 4]
```diff
@@ -790,6 +807,8 @@ public async Task<string[]> GetDistinctMeasurementAttributeValues( ushort key, P
 					featureMatrix.CurrentInterfaceVersion );
 			}
 
+			await CheckLimitResultPerPartFeatureSupport( filter, cancellationToken );
```
[the remark this answers is not in the document]
Comment (czthiele) [reply src/Api.Rest/HttpClient/Data/DataServiceRestClient.cs]: Done

File: src/Api.Rest/HttpClient/Data/DataServiceRestClient.cs [hunk 5]
```diff
@@ -928,6 +947,8 @@ public async Task<DataMeasurementDto[]> GetMeasurementValues( PathInformationDto
 				}
 			}
 
+			await CheckLimitResultPerPartFeatureSupport( filter, cancellationToken );
```
[the remark this answers is not in the document]
Comment (czthiele) [reply src/Api.Rest/HttpClient/Data/DataServiceRestClient.cs]: Done

**Pull request merged**